### PR TITLE
test: port case-insensitive.test.js

### PR DESCRIPTION
**Pull request by rstagi**
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by simoneb** [comment]
@jsumners all these PRs that came in today have been created during an event organized by a couple of folks I know, and they'd like to participate in Hacktoberfest. Can we add the "hacktoberfest-accepted" label to all of them? I don't have permissions to create the label in the repo.

/cc @mcollina

File: test/case-insensitive.test.js [hunk 1]
```diff
@@ -1,120 +1,123 @@
...
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/FOO'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepEqual(JSON.parse(body), {
```
Comment (gurgunday) [inline test/case-insensitive.test.js:28]:
```suggestion
      t.assert.deepStrictEqual(JSON.parse(body), {
```

File: test/case-insensitive.test.js [hunk 2]
```diff
@@ -1,120 +1,123 @@
...
     fastify.inject({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/FOO'
     }, (err, response) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(response.payload), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepEqual(JSON.parse(response.payload), {
```
Comment (gurgunday) [inline test/case-insensitive.test.js:57]:
```suggestion
      t.assert.deepStrictEqual(JSON.parse(response.payload), {
```

File: test/case-insensitive.test.js [hunk 3]
```diff
@@ -1,120 +1,123 @@
...
 
   const fastify = Fastify({
     caseSensitive: false
   })
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
   fastify.get('/foo/:param', (req, reply) => {
-    t.equal(req.params.param, 'bAr')
+    t.assert.strictEqual(req.params.param, 'bAr')
```
Comment (gurgunday) [inline test/case-insensitive.test.js:74]:
```suggestion
    t.assert.deepStrictEqual(req.params.param, 'bAr')
```

File: test/case-insensitive.test.js [hunk 4]
```diff
@@ -1,120 +1,123 @@
...
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/FoO/bAr'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepEqual(JSON.parse(body), {
```
Comment (gurgunday) [inline test/case-insensitive.test.js:87]:
```suggestion
      t.assert.deepStrictEqual(JSON.parse(body), {
```

File: test/case-insensitive.test.js [hunk 5]
```diff
@@ -1,120 +1,123 @@
...
     sget({
       method: 'GET',
       url: 'http://localhost:' + fastify.server.address().port + '/FoO/bAr/baZ'
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepEqual(JSON.parse(body), {
```
Comment (gurgunday) [inline test/case-insensitive.test.js:117]:
```suggestion
      t.assert.deepStrictEqual(JSON.parse(body), {
```

**Review by gurgunday — CHANGES_REQUESTED** [review]

**Comment by jsumners** [comment]
@simoneb please open a new issue that lists all of the PRs you'd like to see have the label.

**Comment by pmarchini** [comment]
Hey @jsumners, I spoke with @simoneb.
I'll create an issue with the list of PRs ASAP 😁

Thank you very much in advance! 🚀

**Comment by rstagi** [comment]
Closing since I see [another PR migrating this file](https://github.com/fastify/fastify/commit/2ee97ffa57f80be390414585ca8c07ecdaf250bc) has already been merged :)

**Pull request closed without merging**